lib/modules/datasource/maven/util.ts

```diff
@@ -93,13 +94,62 @@ export async function downloadHttpProtocol(
       // istanbul ignore next
       logger.debug({ failedUrl }, 'Unsupported host');
     } else {
-      logger.info({ failedUrl, err }, 'Unknown error');
+      logger.info({ failedUrl, err }, 'Unknown HTTP download error');
     }
     return {};
   }
 }
 
-export async function checkHttpResource(
+function isS3NotFound(err: { name: string; message: string }): boolean {
+  return err.message === 'NotFound' || err.message === 'NoSuchKey';
+}
+
+export async function downloadS3Protocol(
+  pkgUrl: URL | string
+): Promise<string | null> {
+  logger.trace({ url: pkgUrl.toString() }, `Attempting to load S3 dependency`);
+  let body: string;
+  try {
+    const s3Url = parseS3Url(pkgUrl.toString());
+    if (s3Url === null) {
+      // istanbul ignore next
```

Comment:
when this can happen? please add a comment like: 
```suggestion
      // istanbul ignore next: some cause
```

Reply:
I don't think it would ever happen (famous last words). It's a null guard.